localstack/services/ses/provider.py

```diff
@@ -201,7 +224,13 @@ def delete_configuration_set_event_destination(
         # not implemented in moto
         # TODO: contribute upstream?
         backend = get_ses_backend(context)
-        backend.config_set_event_destination.pop(configuration_set_name, None)
+        try:
+            backend.config_set_event_destination.pop(configuration_set_name)
+        except KeyError:
+            raise EventDestinationDoesNotExistException(
+                f"No EventDestination found for {configuration_set_name}"
+            )
+
```

Comment:
You're absolutely right I'm leaving the `backend` in an inconsistent state :facepalm:

I like your suggestion on how to proceed with the deletion, I'll implement that, thanks.

> I'm not sure what happens if you try to delete with a wrong event_destination_name. Would it possible to add a test for that case?

Great idea, thanks!

> I feel like the error message is good in the test because we never set an event destination with a different name first as well, so it works. But I'm not sure!

> Note: it seems it can raise 2 different exceptions:
https://docs.aws.amazon.com/ses/latest/APIReference/API_DeleteConfigurationSetEventDestination.html

Yeah I don't cover the configuration set not existing, i.e. the other exception. I'll add that test as well, thanks!